### DISCUSSION TITLE
In the case of AIR, are there any tool to fix  AAB?

### DISCUSSION THREAD

**Discussion by kanisuki** [airsdk/Adobe-Runtime-Support#2926] [Q&A]
Hello,
I understand that I must rewrite AndroidManifest.xml in the aab or apk
 for reading url of web pages in a server.
I is easy with Android Studio or Eclipse.
But in the case of AIR, what to do?
AndroidManifest.xml is a binary file.
With what tool I rewrite it and compile the aab or apk?

**Comment by ajwfrost** [comment]
The AndroidManifest.xml file gets generated by AIR when you build an AAB/APK file.. so if you want to customise this, or add extra permissions etc, you can do this within your application descriptor file.

See the `manifestAdditions` documentation:
https://airsdk.dev/docs/building/application-descriptor-files/elements/android#manifestadditions

thanks

**Reply by kanisuki** [reply]
Thank you.  To do this I think I can use dex2jar, but can I use apkTool to recompile the changed files to bundle to AAB?

**Reply by FliplineStudios** [reply]
You shouldn't need to actually edit the AndroidManifest.xml and recompile -- whatever you want to add, you make the changes to the AIR app's application descriptor XML file instead, and these are then automatically added into the AndroidManifest.xml file which is created when you package with the AIR SDK:

```
<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<application xmlns="http://ns.adobe.com/air/application/50.2">

   <!-- the usual AIR app tags like id, filename, initialWindow go here... -->

   <android>
      <manifestAdditions>
          <![CDATA[
                  <manifest android:installLocation="auto">

                         <!-- anything you add here will be added to AndroidManifest.xml -->
                         <!-- (uses-sdk, uses-permission, queries, application: meta-data, activity, etc.) -->

                  </manifest>
          ]]>
      </manifestAdditions>
   </android>

</application>
```

**Comment by kanisuki** [comment]
Thank you. I will try.

**Comment by kanisuki** [comment]
I understood what to do. I really appreciate your advices. Thanks.
